### ISSUE TITLE
Documentation example with Helm and NGINX doesn't seem to work

### ISSUE THREAD

**Issue by rootd00d** [Azure/secrets-store-csi-driver-provider-azure#342]
**What steps did you take and what happened:**

Referring to the following documentation -- https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/master/docs/ingress-tls.md#deploy-ingress-controller

Seems like the example for consuming TLS secrets from NGINX doesn't work.  This is what I tried --
```
helm upgrade foo-release ingress-nginx/ingress-nginx \
--install \
--force \
--wait \
--timeout=5m \
--atomic \
--set defaultBackend.nodeSelector."beta\.kubernetes\.io/os"=linux \
--set controller.replicaCount=2 \
--set-string controller.config.enable-underscores-in-headers="true" \
--set controller.nodeSelector."beta\.kubernetes\.io/os"=linux \
--set controller.service.annotations."service\.beta\.kubernetes\.io/azure-load-balancer-internal"="true" \
--set controller.extraArgs.default-ssl-certificate="default/foo.tls.secrets.dev" \
--set controller.podAnnotations."config\.linkerd\.io/skip-inbound-ports"="80\,443" \
--set controller.podAnnotations."linkerd\.io/inject"="enabled" \
-f - <<EOF
controller:
  extraVolumes:
    - name: secrets-store-inline
      csi:
          driver: secrets-store.csi.k8s.io
          readOnly: true
          volumeAttributes:
            secretProviderClass: "foo-secrets"
  extraVolumeMounts:
    - name: secrets-store-inline
      mountPath: "/mnt/foo-secrets"
      readOnly: true
EOF
```
The result was the secrets volume was not mapped anywhere.

**What did you expect to happen:**
I was expecting this to create the secret objects, etc., but it seems like the volume definitions don't actually make it to the chart.  

**Anything else you would like to add:**
I'm not sure if this is normal behavior for Helm etc.  I have also tried making the EOF arguments include _all_ of the `--set` parameters, but that seemed to leave them not making it to the deployed pods and services either.  Something about the way Helm "merges" these values -- like does it deep merge or blow away the defaults from values.yaml in the chart if a YAML node is found?

**Comment by rootd00d** [comment]
Figured it out -- the indentation I grabbed from the sample is a bit off.

Once I used this, it worked fine --
```
...
-f - <<EOF
controller:
  extraVolumes:
    - name: secrets-store-inline
      csi:
        driver: secrets-store.csi.k8s.io
        readOnly: true
        volumeAttributes:
          secretProviderClass: "foo-secrets"
  extraVolumeMounts:
    - name: secrets-store-inline
      mountPath: "/mnt/foo-secrets"
      readOnly: true
```
Notice the indentation is less for the 4 lines from "driver:" to "secretProviderClass:"

Thanks!

**Comment by aramase** [comment]
Thank you for reporting the issue. I've opened a PR to fix the indentation in the docs.

**Comment by rootd00d** [comment]
Seems like it works, but I actually have to run the command _twice_ for some reason.  I have no idea why.

Using `helm get all <release>` will return the computed values.yaml that was deployed and the extra volumes sections are empty.  Running the exact same installation command _again_ seems to fix the issue, and the volumes are defined in the values.yaml.

This is probably a Helm issue, and not NGINX or Secrets Store CSI driver specific, but I thought I would follow-up on this example.

**Comment by rootd00d** [comment]
Yep -- issue appears to be fixed as of Helm 3.4.2.  I was using an old 3.0-rc2 release (or something) before.  No issue using this example as is now.
